### PR TITLE
Fixes #32425: Resolve backend CodeQL alerts

### LINKED ISSUE #32425: Resolve backend CodeQL alerts 1699, 1700, 1721, and 1905
## Description

Backend code scanning reports four alerts that can cause numeric corruption or unsafe error/input handling:

- [1699](https://github.com/open-metadata/OpenMetadata/security/code-scanning/1699): implicit narrowing while aggregating web analytics session durations
- [1700](https://github.com/open-metadata/OpenMetadata/security/code-scanning/1700): implicit narrowing of logistic regression weighted scores
- [1721](https://github.com/open-metadata/OpenMetadata/security/code-scanning/1721): internal exception details exposed by the feed servlet
- [1905](https://github.com/open-metadata/OpenMetadata/security/code-scanning/1905): application configuration path derived from request input without containment validation

## Expected behavior

- Numeric calculations preserve their intended precision and reject totals outside the existing model boundary.
- Application configuration resources remain contained beneath the `applications/` classpath directory.
- Feed clients receive a generic HTTP 500 response while detailed exceptions remain in server logs.

## Acceptance criteria

- Add regression coverage for each alert.
- Keep changed service classes at or above 90% line coverage.
- Pass Java formatting, compilation, and focused unit tests.

### PR DESCRIPTION
**Pull request by harshach**
### Describe your changes:

Fixes #32425

This PR resolves backend CodeQL alerts [1699](https://github.com/open-metadata/OpenMetadata/security/code-scanning/1699), [1700](https://github.com/open-metadata/OpenMetadata/security/code-scanning/1700), [1721](https://github.com/open-metadata/OpenMetadata/security/code-scanning/1721), and [1905](https://github.com/open-metadata/OpenMetadata/security/code-scanning/1905). It preserves numeric precision, rejects totals outside the existing Integer model boundary, contains application configuration resources beneath `applications/`, and prevents feed responses from exposing internal exception details.

#
### Type of change:

- [x] Bug fix

#
### High-level design:

Numeric accumulators now use the source calculation's required precision, with `Math.toIntExact` applied only at the existing Integer schema boundary so overflow fails instead of silently wrapping.

Application configuration paths are resolved and normalized beneath the `applications/` classpath root, rejected when absolute or outside that root, and passed to a private raw resource reader. A stricter application-name allowlist was avoided to preserve valid existing names and nested resource layouts.

Feed processing keeps full exception details in server logs while returning a fixed HTTP 500 message to clients. No schema, data migration, UI, or rollout changes are required.

#
#### Use cases covered

- Fractional classifier weights affect the predicted severity correctly.
- Valid web-analytics sessions aggregate durations and update workflow statistics.
- Session totals outside the Integer model range fail instead of wrapping.
- Valid application configurations continue loading with environment substitution.
- Parent traversal, nested traversal, absolute paths, and missing application names are rejected.
- Valid feed requests delegate to Engine.IO.
- Feed failures return a generic HTTP 500 without exposing internal details.

#### Unit tests

- [x] I added unit tests for the new and changed logic.
- Files added or updated:
  - `LogisticRegressionIncidentSeverityClassifierTest.java`
  - `WebAnalyticsUserActivityAggregatorTest.java`
  - `ConfigurationReaderTest.java`
  - `FeedServletTest.java`
- Changed-class line coverage:
  - `ConfigurationReader`: 100.0%
  - `WebAnalyticsUserActivityAggregator`: 100.0%
  - `FeedServlet`: 100.0%
  - `LogisticRegressionIncidentSeverityClassifier`: 90.9%

#### Backend integration tests

- [x] Not applicable — no REST resource contract or entity behavior changed.

#### Ingestion integration tests

- [x] Not applicable — no ingestion changes.

#### Playwright (UI) tests

- [x] Not applicable — no UI changes.

#### Manual testing performed

1. Ran `mvn test -pl openmetadata-service -Pstatic-code-analysis -Dtest=LogisticRegressionIncidentSeverityClassifierTest,WebAnalyticsUserActivityAggregatorTest,ConfigurationReaderTest,FeedServletTest` and verified 19 tests passed.
2. Ran `mvn test-compile -pl openmetadata-service` and verified the module compiled successfully.
3. Ran `mvn -pl openmetadata-service spotless:check` and verified all Java files were clean.
4. Generated the JaCoCo report and verified every changed service class met the 90% line-coverage target.

#
### UI screen recording / screenshots:

Not applicable — no UI changes.

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developers/contribute) document.
- [x] My PR title is `Fixes <issue-number>: <short explanation>`.
- [x] My PR is linked to a GitHub issue via `Fixes #<issue-number>` above.
- [x] Hard-to-understand behavior is documented where applicable.
- [x] JSON Schema changes are not applicable.
- [x] UI recordings are not applicable.
- [x] I have added tests and listed them above.
- [x] I have added tests that cover the exact scenarios being fixed.

### CODE REVIEW

File: openmetadata-service/src/main/java/org/openmetadata/service/util/incidentSeverityClassifier/LogisticRegressionIncidentSeverityClassifier.java [hunk 1]
```diff
@@ -58,7 +58,7 @@ private double[] dotProduct(double[] vectorX) {
     // compute the dot product of the input vector and the coef. matrix
     double[] result = new double[coefMatrix[0].length];
     for (int i = 0; i < coefMatrix.length; i++) {
-      int sum = 0;
+      double sum = 0;
       for (int j = 0; j < vectorX.length; j++) {
         sum += vectorX[j] * coefMatrix[j][i];
       }
```
[the remark this answers is not in the document]
Comment (harshach) [reply openmetadata-service/src/main/java/org/openmetadata/service/util/incidentSeverityClassifier/LogisticRegressionIncidentSeverityClassifier.java]: Fixed in 47bb76aac2. dotProduct now iterates over coefMatrix[0].length, so it computes all five class logits. I tightened the Tier 5 regression to assert Severity5; it failed before the fix with Severity4 and passes afterward. All 19 focused PR tests and Spotless check pass.

**Review by mohityadav766 — APPROVED** [review]
